### PR TITLE
fix: sweep marketplaces

### PR DESCRIPTION
**Pull request by JackShort**
* allowing you to sweep specific marketplaces

### CODE REVIEW

File: src/nft/components/collection/Sweep.tsx [hunk 1]
```diff
@@ -206,7 +210,14 @@ export const Sweep = ({ contractAddress, minPrice, maxPrice }: SweepProps) => {
       (asset) => BigNumber.from(asset.priceInfo.ETHPrice).gte(0) && !asset.susFlag
     )
 
-    validAssets = validAssets.slice(0, Math.max(collectionAssets.length, nftxAssets.length, nft20Assets.length))
+    validAssets = validAssets.slice(
+      0,
+      Math.max(
+        collectionAssets ? collectionAssets.length : 0,
```
Comment (vm) [inline src/nft/components/collection/Sweep.tsx:216]: collectionAssets?.length ?? 0

File: src/nft/components/collection/Sweep.tsx [hunk 2]
```diff
@@ -413,7 +424,7 @@ export function useSweepFetcherParams(
       if (market === 'others') {
         return { contractAddress, traits, markets }
       }
-      if (!markets.includes(market)) return { contractAddress: '', traits: [], markets: [] }
```
Comment (vm) [inline src/nft/components/collection/Sweep.tsx:416]: why this change?

File: src/nft/components/collection/Sweep.tsx [hunk 3]
```diff
@@ -177,13 +177,17 @@ export const Sweep = ({ contractAddress, minPrice, maxPrice }: SweepProps) => {
   const nft20Assets = useLazyLoadSweepAssetsQuery(nft20Params)
 
   const { sortedAssets, sortedAssetsTotalEth } = useMemo(() => {
-    if (!collectionAssets || !nftxAssets || !nft20Assets)
+    if (!collectionAssets && !nftxAssets && !nft20Assets) {
```
Comment (vm) [inline src/nft/components/collection/Sweep.tsx:180]: why this change?

File: src/nft/components/collection/Sweep.tsx (hunk 2 above)
Comment (JackShort) [reply src/nft/components/collection/Sweep.tsx:416]: if its being filtered its already being passed as a parameter when market === `'others'`. This prevents it from happening twice

File: src/nft/components/collection/Sweep.tsx (hunk 3 above)
Comment (JackShort) [reply src/nft/components/collection/Sweep.tsx:180]: if any markets were selected it would return undefined because nftxAssets or nft20 assets would be undefined

**Review by vm — APPROVED** [review]
